proto/autokitteh/triggers/v1/trigger.proto

```diff
@@ -3,16 +3,26 @@ syntax = "proto3";
 package autokitteh.triggers.v1;
 
 import "autokitteh/program/v1/program.proto";
-import "autokitteh/values/v1/values.proto";
-import "buf/validate/validate.proto";
 
 message Trigger {
+  enum SourceType {
+    SOURCE_TYPE_UNSPECIFIED = 0;
+    SOURCE_TYPE_CONNECTION = 1;
+    SOURCE_TYPE_WEBHOOK = 2;
+    SOURCE_TYPE_SCHEDULE = 3;
+  }
+
   string trigger_id = 1;
   string name = 2;
-  string connection_id = 3;
-  string env_id = 4; // if empty, applies to all envs.
+  SourceType source_type = 3;
+  string env_id = 4;
   string event_type = 5;
-  program.v1.CodeLocation code_location = 6 [(buf.validate.field).required = true];
+  program.v1.CodeLocation code_location = 6;
   string filter = 7;
-  map<string, values.v1.Value> data = 8;
+
+  string connection_id = 50; // if source_type == CONNECTION.
+  string schedule = 51; // if source_type == SCHEDULE.
+
+  // read only.
+  string webhook_slug = 100; // if source_type == WEBHOOK, after creation.
```

Comment:
`SourceType` is just an enum, not a message, so the value can't be added there. As for `oneof` - I personally prefer to avoid it, it complicates generated code and limits future extensibility. But in that case it does make sense to add validation checks:
1. If source_type = CONNECTION then `connection_id` isn't empty, but `schedule` and `webhook_slug` are
2. IF source_type = WEBHOOK, ...
3. IF source_type = SCHEDULE, ...

Reply:
i'd rather have these separated. if we add anything in the future that requires a different type. also, webhook_slug is generated, schedule is not.

Reply:
@itayd I agree with you, but see my note about validation checks.

Reply:
done